src/chainrules.jl

```diff
@@ -48,14 +49,34 @@ function ChainRulesCore.rrule(
             block_rows = row_idxs[i]:(row_idxs[i] + nrows[i] - 1)
             block_cols = col_idxs[i]:(col_idxs[i] + ncols[i] - 1)
             return InplaceableThunk(
+                X̄ -> mul!(X̄, ȳ[block_rows], v[block_cols]', true, true),
                 @thunk(ȳ[block_rows] * v[block_cols]'),
-                X̄ -> mul!(X̄, ȳ[block_rows], v[block_cols]', true, true)
             )
         end
 
         b̄m = Tangent{BlockDiagonal{T, V}}(;blocks=Δblocks)
-        v̄ = InplaceableThunk(@thunk(bm' * ȳ), X̄ -> mul!(X̄, bm', ȳ, true, true))
+        v̄ = InplaceableThunk(X̄ -> mul!(X̄, bm', ȳ, true, true), @thunk(bm' * ȳ))
         return NoTangent(), b̄m, v̄
     end
     return y, bm_vector_mul_pullback
 end
+
+function ProjectTo(b::BlockDiagonal)
```

Comment:
Do we have a canonical way to test methods of `ProjectTo`?

Reply:
We don't have a test utility but I should probably add some tests